yt_dlp/extractor/tumblr.py

```diff
@@ -107,116 +237,170 @@ class TumblrIE(InfoExtractor):
+
+        if 'url' not in media_json and not video_url and not iframe_url:
+            # external video host (but we weren't able to figure it out from the api)
             iframe_url = self._search_regex(
                 r'src=["\'](https?://safe\.txmblr\.com/svc/embed/inline/[^"\']+)["\']',
                 webpage, 'embed iframe url', default=None)
             return self.url_result(iframe_url or redirect_url, 'Generic')
 
-        iframe = self._download_webpage(
-            iframe_url, video_id, 'Downloading iframe page',
-            headers={'Referer': redirect_url})
-
-        duration = None
-        sources = []
-
-        sd_url = self._search_regex(
-            r'<source[^>]+src=(["\'])(?P<url>.+?)\1', iframe,
-            'sd video url', default=None, group='url')
-        if sd_url:
-            sources.append((sd_url, 'sd'))
-
-        options = self._parse_json(
-            self._search_regex(
-                r'data-crt-options=(["\'])(?P<options>.+?)\1', iframe,
-                'hd video url', default='', group='options'),
-            video_id, fatal=False)
-        if options:
-            duration = int_or_none(options.get('duration'))
-            hd_url = options.get('hdUrl')
-            if hd_url:
-                sources.append((hd_url, 'hd'))
-
-        formats = [{
-            'url': video_url,
-            'ext': 'mp4',
-            'format_id': format_id,
-            'height': int_or_none(self._search_regex(
-                r'/(\d{3,4})$', video_url, 'height', default=None)),
-            'quality': quality,
-        } for quality, (video_url, format_id) in enumerate(sources)]
-
+        formats = formats or [{
+            'url': media_json.get('url', video_url),
+            'format_id': '0',
+            'width': media_json.get('width', int_or_none(
+                self._og_search_property('video:width', webpage, default=None))),
+            'height': media_json.get('height', int_or_none(
+                self._og_search_property('video:height', webpage, default=None))),
+        }]
         self._sort_formats(formats)
 
-        # The only place where you can get a title, it's not complete,
-        # but searching in other places doesn't work for all videos
-        video_title = self._html_search_regex(
-            r'(?s)<title>(?P<title>.*?)(?: \| Tumblr)?</title>',
-            webpage, 'title')
+        # the url we're extracting from might be an original post or it might be a reblog.
+        # if it's a reblog, og:description will be the reblogger's comment, not the uploader's.
+        # content_json is the op's post, so if it exists but has no text, there's no description
+        description = ('\n\n'.join(
+            (item.get('text') for item in content_json if item.get('type') == 'text'))
+            or (None if content_json else self._og_search_description(webpage, default=None)))
+        uploader_id = traverse_obj(post_json, ('reblogged_root_name'), ('blog_name'))
```

Comment:
why do u put brackets everywhere!!?
```suggestion
        uploader_id = traverse_obj(post_json, 'reblogged_root_name', 'blog_name')
```